distribution/obproxy/start.sh

```diff
@@ -1,21 +1,57 @@
 #!/bin/bash
 
-
 if [ -z $APP_NAME ]; then
     echo "env variable APP_NAME is required"
     exit 1
 fi
 
 if [ -z $PROXYRO_PASSWORD_HASH ]; then
-    PROXYRO_PASSWORD_HASH=`echo -n "$PROXYRO_PASSWORD" | sha1sum | awk '{print $1}'`
+    PROXYRO_PASSWORD_HASH=$(echo -n "$PROXYRO_PASSWORD" | sha1sum | awk '{print $1}')
 fi
 
+if [ -z $PROXYSYS_PASSWORD ]; then
+    echo "env variable PROXYSYS_PASSWORD is required"
+    exit 1
+fi
+
+PROXYSYS_PASSWORD_HASH=$(echo -n "$PROXYSYS_PASSWORD" | sha1sum | awk '{print $1}')
+
+opts="obproxy_sys_password=$PROXYSYS_PASSWORD_HASH"
+
+function concat_opts {
+    if [ -z "$1" ]; then
+        echo $2
+    elif [ -z "$2" ]; then
+        echo $1
+    else
+        echo "$1,$2"
+    fi
+}
+
+[ -z "$ODP_PROMETHEUS_SYNC_INTERVAL" ] && opts=$(concat_opts $opts "prometheus_sync_interval=1s")
+[ -z "$ODP_ENABLE_METADB_USED" ] && opts=$(concat_opts $opts "enable_metadb_used=false")
+[ -z "$ODP_SKIP_PROXY_SYS_PRIVATE_CHECK" ] && opts=$(concat_opts $opts "skip_proxy_sys_private_check=true")
+[ -z "$ODP_LOG_DIR_SIZE_THRESHOLD" ] && opts=$(concat_opts $opts "log_dir_size_threshold=10G")
+[ -z "$ODP_ENABLE_PROXY_SCRAMBLE" ] && opts=$(concat_opts $opts "enable_proxy_scramble=true")
+[ -z "$ODP_ENABLE_STRICT_KERNEL_RELEASE" ] && opts=$(concat_opts $opts "enable_strict_kernel_release=false")
+
+while IFS='=' read -r key value; do
+    # If the key has prefix "ODP_" then add it to the opts
+    if [[ $key == ODP_* ]]; then
+        # Remove the prefix "ODP_" from the key and transform to lower case
+        key=$(echo $key | sed 's/^ODP_//g' | tr '[:upper:]' '[:lower:]')
+        opts=$(concat_opts $opts "$(printf "%s=%s" "$key" "$value")")
```

Comment:
it's better to add a quote for value, some config values may contain special characters such as '&' which will cause the command behave unexpectedly

Reply:
You mean `opts=$(concat_opts $opts "$(printf "%s=\"%s\"" "$key" "$value")")`?

Reply:
yes